### DISCUSSION TITLE
Return a custom JSON when defining a new bouncer

### DISCUSSION THREAD

**Discussion by 3adel-bassiony** [adonisjs/core#3884] [Help] [answered]
I'm using the @Adonisjs/bouncer and I can't find anything in the documentation to return custom JSON or how I can handle the multi-lang using i18n
It's working well but I want to return a custom JSON object in case the user is not authorized to make the action
The only thing mentioned in the docs is: 

```
export const { actions } = Bouncer
  .define('viewPost', (user: User, post: Post) => {
    if (post.userId === user.id || post.status === 'published') {
      return true
    }

    return Bouncer.deny('Post not found', 404)
  })
```

But I can't return a custom JSON or handle the multi-lang

**Comment by thetutlage** [comment, marked as answer]
Bouncer raises the `AuthorizationException` with `E_AUTHORIZATION_FAILURE` error code. You can simply capture this exception inside the global exception handler and return whatever you like from there.

Trimmed down version of Exception handler

```ts
handle(error, ctx) {
  if(error.code === 'E_AUTHORIZATION_FAILURE') {
    ctx.response.status(error.status).send({ message: error.message })
  }
}
```

**Reply by 3adel-bassiony** [reply]
Yea, I added it inside the `app/Exceptions/Handler.ts` and it worked 🎉
